Stack/Opc.Ua.Core/Types/BuiltIn/NodeId.cs

```diff
@@ -663,6 +664,12 @@ public static NodeId Parse(string text)
                     return new NodeId(Convert.FromBase64String(text.Substring(2)), namespaceIndex);
                 }
 
+                // parse the namespace index if present.
+                if (text.StartsWith("nsu=", StringComparison.Ordinal))
+                {
+                    throw new ServiceResultException(StatusCodes.BadNodeIdInvalid, "Invalid namespace definition as Uri for NodeId.");
```

Comment:
return argumentexception to pint this out as error in the calling code that should use a expandednodeid here.